pkg/controller/route/routecontroller.go

```diff
@@ -26,73 +26,140 @@ import (
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/cache"
 	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/util/metrics"
 	nodeutil "k8s.io/kubernetes/pkg/util/node"
+	utilruntime "k8s.io/kubernetes/pkg/util/runtime"
 	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+	"k8s.io/kubernetes/pkg/watch"
 )
 
+type StatusType string
+
 const (
 	// Maximal number of concurrent CreateRoute API calls.
 	// TODO: This should be per-provider.
 	maxConcurrentRouteCreations int = 200
 	// Maximum number of retries of route creations.
 	maxRetries int = 5
 	// Maximum number of retries of node status update.
-	updateNodeStatusMaxRetries int = 3
+	updateNodeStatusMaxRetries int        = 3
+	statusDelete               StatusType = "Delete"
+	statusAdd                  StatusType = "Add"
 )
 
 type RouteController struct {
 	routes      cloudprovider.Routes
 	kubeClient  clientset.Interface
 	clusterName string
 	clusterCIDR *net.IPNet
+	// Node framework and store
+	nodeController *framework.Controller
```

Comment:
I don't really like one controller depending on the other one. I don't think we have this pattern anywhere else. @davidopp